src/content_sao.cpp

```diff
@@ -1015,9 +1015,11 @@ void PlayerSAO::step(float dtime, bool send_recommended)
 		// Get nose/mouth position, approximate with eye position
 		v3s16 p = floatToInt(getEyePosition(), BS);
 		MapNode n = m_env->getMap().getNodeNoEx(p);
+		content_t content = n.getContent();
 		const ContentFeatures &c = m_env->getGameDef()->ndef()->get(n);
-		// If player is alive & no drowning, breathe
-		if (m_hp > 0 && m_breath < m_prop.breath_max && c.drowning == 0)
+		// If player is alive & no drowning & not in ignore, breathe
+		if (content != CONTENT_IGNORE && m_hp > 0 &&
```

Comment:
Can this line be re-arranged to short-circuit as early as possible? So terms most likely to be false earliest.
Not being in ignore is probably the least likely to be false so should probably be last.
Best order seems ot be:
m_breath < m_prop.breath_max
c.drowning == 0
content != CONTENT_IGNORE
m_hp > 0

Reply:
Done thanks.